packages/labs/analyzer/src/test/lit-element/events_test.ts

```diff
@@ -73,24 +78,55 @@ test('Event with type', ({element}) => {
   const event = element.events.get('typed-event');
   assert.ok(event);
   assert.equal(event.name, 'typed-event');
-  assert.equal(event.typeString, 'MouseEvent');
+  assert.equal(event.type?.text, 'MouseEvent');
   assert.equal(event.description, undefined);
+  assert.equal(event.type?.references[0].name, 'MouseEvent');
+  assert.equal(event.type?.references[0].isGlobal, true);
 });
 
 test('Event with type and description', ({element}) => {
   const event = element.events.get('typed-event-two');
   assert.ok(event);
   assert.equal(event.name, 'typed-event-two');
-  assert.equal(event.typeString, 'MouseEvent');
+  assert.equal(event.type?.text, 'MouseEvent');
   assert.equal(event.description, 'This is a typed event');
 });
 
 test('Event with type and dash-separated description', ({element}) => {
   const event = element.events.get('typed-event-three');
   assert.ok(event);
   assert.equal(event.name, 'typed-event-three');
-  assert.equal(event.typeString, 'MouseEvent');
+  assert.equal(event.type?.text, 'MouseEvent');
   assert.equal(event.description, 'This is another typed event');
 });
 
+test('Event with local custom event type', ({element}) => {
+  const event = element.events.get('local-custom-event');
+  assert.ok(event);
+  assert.equal(event.type?.text, 'LocalCustomEvent');
+  assert.equal(event.type?.references[0].package, '@lit-internal/test-events');
+  assert.equal(event.type?.references[0].module, 'element-a.js');
+  assert.equal(event.type?.references[0].name, 'LocalCustomEvent');
+});
+
+test('Event with imported custom event type', ({element}) => {
+  const event = element.events.get('external-custom-event');
+  assert.ok(event);
+  assert.equal(event.type?.text, 'ExternalCustomEvent');
+  assert.equal(event.type?.references[0].package, '@lit-internal/test-events');
+  assert.equal(event.type?.references[0].module, 'custom-event.js');
+  assert.equal(event.type?.references[0].name, 'ExternalCustomEvent');
+});
+
+test('Event with generic custom event', ({element}) => {
+  const event = element.events.get('generic-custom-event');
+  assert.ok(event);
+  assert.equal(event.type?.text, 'CustomEvent<ExternalClass>');
```

Comment:
Do you want to test a more complex type? I could see an inline detail type being not-uncommon, like:

```ts
CustomEvent<{name: string, age?: number}>
```

Reply:
So I added this test, but for some reason the `typedoc` visitor never gets to type references inside the interface. Commented out some asserts in the test and added TODO w/issue for replacing the Type visitor: https://github.com/lit/lit/issues/3001